consensus/src/quorum_store/batch_generator.rs

```diff
@@ -99,112 +96,79 @@ impl BatchGenerator {
         }
     }
 
-    pub(crate) async fn handle_scheduled_pull(&mut self, max_count: u64) -> Option<Fragment> {
+    pub(crate) async fn handle_scheduled_pull(&mut self, max_count: u64) -> Option<Batch> {
         // TODO: as an optimization, we could filter out the txns that have expired
 
-        let mut exclude_txns: Vec<_> = self
+        let exclude_txns: Vec<_> = self
             .batches_in_progress
             .values()
             .flatten()
             .cloned()
             .collect();
-        exclude_txns.extend(self.batch_builder.summaries().clone());
 
         trace!("QS: excluding txs len: {:?}", exclude_txns.len());
-        let mut end_batch = false;
         let pulled_txns = self
             .mempool_proxy
             .pull_internal(
                 max_count,
                 self.config.mempool_txn_pull_max_bytes,
-                // allow creating non-full fragments
-                // is this a good place to disable fragments actually?
                 true,
```

Comment:
basically this should be true only if we want to create a partial batch, false otherwise.